### DISCUSSION TITLE
Plz update with-mobx-lite SSR.

### DISCUSSION THREAD

**Discussion by Kir93** [vercel/next.js#15085] [Help]
# Feature request

## Is your feature request related to a problem? Please describe.

I want with-mobx-lite SSR. but, I didnt find SSR example.

## Describe the solution you'd like

with-mobx-lite SSR example.

**Comment by jamesmosier** [comment]
Could you elaborate a bit more? The [with-mobx-react-lite](https://github.com/vercel/next.js/tree/canary/examples/with-mobx-react-lite) example works with server rendering. 

Additionally, feel free to contribute a PR for this!

**Reply by jamesmosier** [reply]
This example works with server rendering out of the box. To hydrate the store, you'll see in [`_app.js`](https://github.com/vercel/next.js/blob/canary/examples/with-mobx-react-lite/pages/_app.js) it says:

https://github.com/vercel/next.js/blob/9b711883cfebfce0997cd9fe188a33cd433e9981/examples/with-mobx-react-lite/pages/_app.js#L4-L5

Therefore in any of your pages you could do:

```js
export function getServerSideProps() {
  return {
    props: { initialStoreData: 'your data for the store goes here' }
  }
}
```

**Reply by Kir93** [reply]
I didn't understand about running the function.
example,

store/index.js
```
function initializeData(initialData = store || {}) {
    const { lastUpdate = Date.now(), light, login, logout, loadInfo = false } = initialData;
    return {
        lastUpdate,
        light: Boolean(light),
        login() {
            return (loadInfo = true);
        },
        logout() {
            return (loadInfo = false);
        },
        loadInfo,
    };
}

```
I tried SSR in index.js

```
import { store } from '../store';
import { useObserver } from 'mobx-react';

export default function Index() {
    return useObserver(() => <div>{store.loadInfo ? <div>Login</div> : <div>Logout</div>}</div>);
}

export function getServerSideProps() {
    const result = login();
    return { props: { initialStoreData: { loadInfo: result } } };
}
```

But, This didn't work.

Where did I misunderstand?

I'll wait for a good answer :)

**Comment by Kir93** [comment]
I dont found using getServerSidePorps example.
Can you provide getServerSidePorps example?
